examples/tf/dask_script.py

```diff
@@ -0,0 +1,73 @@
+import click
+from dask.distributed import Client
+from dask_jobqueue import SLURMCluster
+
+from mnist_example import train_dense_model
+
+
+@click.command()
+@click.option(
+    'n_gpus',
+    '-n',
+    default=1,
+    help='The number of GPUs on which to run the mnist examples in parallel. Defaults to 1.',
+    type=int,
+)
+@click.option(
+    'save',
+    '-s',
+    '--save',
+    is_flag=True,
+    help='Whether you want to save the models or not',
+)
+def launch_dask_tasks(n_gpus, save):
+    assert 0 < n_gpus < 5, 'You need to request between 1 and 4 GPUs.'
+
+    job_name = 'dask_mnist_tf_example'
+    if n_gpus > 1:
+        job_name += '_multi_gpus'
+
+    cluster = SLURMCluster(
+        n_workers=n_gpus,
+        cores=n_gpus,
+        job_cpu=10,
+        memory='10GB',
+        job_name=job_name,
+        walltime='1:00:00',
+        interface='ib0',
+        job_extra=[
+            f'--gres=gpu:{n_gpus}',
+            '--qos=qos_gpu-dev',
+            '--distribution=block:block',
+            '--hint=nomultithread',
+            '--output=%x_%j.out',
+        ],
+        env_extra=[
+            'module purge',
+            'module load tensorflow-gpu/py3/2.1.0',
+        ],
+        extra=[f'--resources GPU={n_gpus}'],
+    )
+
+    print(cluster.job_script())
+
+    client = Client(cluster)
+    futures = [client.submit(
+        # function to execute
+        train_dense_model,
+        # *args
+        None, save,
+        # this function has potential side effects
+        pure=not save,
```

Comment:
Out of curiosity again, can you elaborate a bit more why this function has potential side-effects ?

Reply:
so I took the definition of a pure function from [here](https://www.sitepoint.com/functional-programming-pure-functions/), and in their understanding, a side-effect could be something like writing a log (or this case the weights of an NN).

Reply:
OK fair enough, I think you read the doc too much ...  `pure` only matter if you are using `.submit` with the same function and the same arguments. Dask has some caching mechanism where it may not recompute (see https://docs.dask.org/en/latest/caching.html) the results.

But now I am a bit confused, I thought the function `train_dense_model` would be called with different parameters, e.g. learning rate and it seems like it is called with the same parameters ... is there a use case for this? Am I missing something?

Reply:
Oh yes this is purely for example's sake, in real life you would call it with different arguments indeed, I just didn't want to complicate things too much but I can indeed try and have different arguments if you think it'd be clearer.

Reply:
Different arguments would be better IMO because it would look more like a real-life example and easier to relate to as a DL practicioner